### PR TITLE
Autocomplete: Use tabby style helpers for completion strings in tests

### PR DESCRIPTION
**Pull request by philipp-spiess**
c.f. https://github.com/sourcegraph/cody/pull/393

This changes the completion examples in the docs once again to match the style of tabby: https://sourcegraph.com/github.com/TabbyML/tabby/-/blob/clients/tabby-agent/src/postprocess/testUtils.ts?L22:17-22:23

## Test plan

- Tests pass

### CODE REVIEW

File: vscode/src/completions/completion.test.ts [hunk 1]
```diff
@@ -114,7 +133,7 @@ describe('Cody completions', () => {
             })
 
             if (!code.includes(CURSOR_MARKER)) {
-                throw new Error('The test code must include a | to denote the cursor position')
+                throw new Error('The test code must include a ║ to denote the cursor position')
```
Comment (dominiccooney) [inline vscode/src/completions/completion.test.ts:136]: May as well go all in and use `CURSOR_MARKER` here.
```suggestion
                throw new Error(`The test code must include a ${CURSOR_MARKER} to denote the cursor position`)
```

File: vscode/src/completions/completion.test.ts [hunk 2]
```diff
@@ -187,7 +206,7 @@ describe('Cody completions', () => {
                 public end: Position
 
                 constructor(startLine: number, startCharacter: number, endLine: number, endCharacter: number) {
-                    this.startLine = ${CURSOR_MARKER}
+                    this.startLine = ║
```
Comment (dominiccooney) [inline vscode/src/completions/completion.test.ts:209]: Despite my earlier comment about the error message string literal, I think it looks good to write it as you have it here. The test is nicely readable this way.

File: vscode/src/completions/completion.test.ts [hunk 3]
```diff
@@ -236,29 +249,31 @@ describe('Cody completions', () => {
...
         }
         `,
-            [createCompletionResponse('if (true) {\n        console.log(3);\n    }\n    console.log(4);')]
+            [
+                completion`
+                    ├if (true) {
+                        console.log(3);
+                    }
+                    console.log(4);┤
+                ┴┴┴┴`,
```
Comment (dominiccooney) [inline vscode/src/completions/completion.test.ts:261]: What does it mean to have four of these? Anywhere is OK, or there should be a selection, or...?

**Review by dominiccooney — APPROVED** [review]
Looks great, a bunch of feedback inline. Some style advice I leave to your taste... The bit about the start offset needs attention before merging I think.

File: vscode/src/completions/completion.test.ts (hunk 1 above)
Comment (philipp-spiess) [reply vscode/src/completions/completion.test.ts:136]: I think that's a good idea given that this message printed `|` which we haven't been using for over a month lol

File: vscode/src/completions/completion.test.ts (hunk 3 above)
Comment (philipp-spiess) [reply vscode/src/completions/completion.test.ts:261]: These characters are used to denote that everything is indented by 4 characters (so the unindent logic only expands until the first occurence). It doesn't matter whether its one or four. Tabby uses four so you can also easily count them and see "4 spaces" I think 🙂

File: vscode/src/completions/completion.test.ts [hunk 4]
```diff
@@ -15,6 +15,14 @@ import { CompletionsCache } from './cache'
 import { History } from './history'
 import { createProviderConfig } from './providers/anthropic'
 
+const CURSOR_MARKER = '║'
```
[the remark this answers is not in the document]
Comment (philipp-spiess) [reply vscode/src/completions/completion.test.ts:18]: Makes sense, prefer the full block too!

**Pull request merged**